Source/Core/AxisAlignedBoundingBox.js

```diff
@@ -191,6 +193,28 @@ define([
         return Intersect.INTERSECTING;
     };
 
+    /**
+     * If the given position is not already within the box, projects the given position onto the box
+     * @param {Cartesian3} position The position being projected onto this AxisAlignedBoundingBox.
+     * @param {Cartesian3} [result] The object onto which to store the result.
+     * @returns {Cartesian3} A projected version of the inputted position if it was not originally within the AxisAlignedBoundingBox.
+     */
+    AxisAlignedBoundingBox.prototype.clampToBounds = function(position, result) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.object('position', position);
+        //>>includeEnd('debug');
+
+        result = position.clone(result);
```

Comment:
Use `result = Cartesian3.clone(position, result)` instead.  We always prefer to use the static function for math classes like `Cartesian3`.  See https://github.com/AnalyticalGraphicsInc/cesium/tree/master/Documentation/Contributors/CodingGuide#use-prototype-functions-for-fundamental-classes-sparingly

Reply:
Same throughout, you have this in a few different places